x-pack/plugins/ml/public/embeddables/lens/lens_vis_layer_selection_flyout/layer/compatible_layer.tsx

```diff
@@ -0,0 +1,432 @@
+import { extractErrorMessage } from '../../../../../common/util/errors';
+import { MlApiServices } from '../../../../application/services/ml_api_service';
+import { basicJobValidation } from '../../../../../common/util/job_utils';
+import { JOB_ID_MAX_LENGTH } from '../../../../../common/constants/validation';
+import { invalidTimeIntervalMessage } from '../../../../application/jobs/new_job/common/job_validator/util';
+import { ML_APP_LOCATOR } from '../../../../../common/constants/locator';
+
+interface Props {
+  layer: LayerResult;
+  layerIndex: number;
+  embeddable: Embeddable;
+  share: SharePluginStart;
+  data: DataPublicPluginStart;
+  application: ApplicationStart;
+  kibanaConfig: IUiSettingsClient;
+  ml: MlApiServices;
+}
+
+enum STATE {
+  DEFAULT,
+  VALIDATING,
+  SAVING,
+  SAVE_SUCCESS,
+  SAVE_FAILED,
+}
+
+export const CompatibleLayer: FC<Props> = ({
+  layer,
+  layerIndex,
+  embeddable,
+  share,
+  data,
+  application,
+  ml,
+  kibanaConfig,
+}) => {
+  const [jobId, setJobId] = useState('');
+  const [startJob, setStartJob] = useState(true);
+  const [runInRealTime, setRunInRealTime] = useState(true);
+  const [bucketSpan, setBucketSpan] = useState(DEFAULT_BUCKET_SPAN);
+
+  const [jobIdValid, setJobIdValid] = useState<string>('');
+  const [bucketSpanValid, setBucketSpanValid] = useState<string>('');
+  const [state, setState] = useState<STATE>(STATE.DEFAULT);
+  const [createError, setCreateError] = useState<{ text: string; errorText: string } | null>(null);
+
+  function createADJobInWizard() {
+    convertLensToADJob(embeddable, share, layerIndex);
+  }
+
+  async function createADJob() {
+    setState(STATE.SAVING);
+    setCreateError(null);
+    const result = await createAndSaveJob(
+      jobId,
+      bucketSpan,
+      embeddable,
+      startJob,
+      runInRealTime,
+      data.dataViews,
+      kibanaConfig,
+      data.query.timefilter.timefilter,
+      ml,
+      layerIndex
+    );
+    const error = checkForCreationErrors(result);
+    if (error === null) {
+      setState(STATE.SAVE_SUCCESS);
+    } else {
+      setState(STATE.SAVE_FAILED);
+      setCreateError(error);
+    }
+  }
+
+  const viewResults = useCallback(
+    async (jobType: CREATED_BY_LABEL | null) => {
+      const { timeRange } = embeddable.getInput();
+      const page = jobType === CREATED_BY_LABEL.MULTI_METRIC ? 'explorer' : 'timeseriesexplorer';
+      const locator = share.url.locators.get(ML_APP_LOCATOR);
+      if (locator) {
+        const url = await locator!.getUrl({
+          page,
+          pageState: {
+            jobIds: [jobId],
+            timeRange,
+          },
+        });
+
+        application.navigateToUrl(url);
+      }
+    },
+    [jobId, embeddable, share, application]
+  );
+
+  function setStartJobWrapper(start: boolean) {
+    setStartJob(start);
+    setRunInRealTime(start && runInRealTime);
+  }
+
+  useDebounce(
+    async function validateJobId() {
```

Comment:
not sure if you're supposed to use `useDebounce` with async functions

Reply:
Updated in https://github.com/elastic/kibana/pull/136421/commits/ae35f0de4bed8502f2c0f97cacee58d0675c6f29